src/llmcompressor/typing.py

```diff
@@ -1,17 +1,27 @@
-from typing import Union
+from typing import TYPE_CHECKING, Callable, Union
 
+import torch
 from datasets import Dataset, DatasetDict, IterableDataset
 from transformers import (
     BaseImageProcessor,
     FeatureExtractionMixin,
+    PreTrainedModel,
     PreTrainedTokenizer,
     ProcessorMixin,
 )
 
+if TYPE_CHECKING:
+    from llmcompressor.args.dataset_arguments import DatasetArguments
+
 # Tokenizer or Processor. Processors do not inherit from a unified base class
 Processor = Union[
     PreTrainedTokenizer, BaseImageProcessor, FeatureExtractionMixin, ProcessorMixin
 ]
 
 # Supported dataset types, IterableDataset is a streamed dataset
 DatasetType = Union[Dataset, DatasetDict, IterableDataset]
+
+# Pipeline callable
+PipelineFn = Callable[
+    [PreTrainedModel, torch.utils.data.DataLoader, "DatasetArguments"], None
+]
```

Comment:
nit: feels awkward to me, I know we discussed an abstract base class as being too much for this, but maybe a `Protocol` would be a nice middle-ground

```python
class Pipeline(Protocol):
    def run_pipeline(self, model: PreTrainedModel, data_loader: torch.utils.data.DataLoader, dataset_args: "DatasetArguments") -> None:
        ...
```

Reply:
I used the ABC